app/javascript/controllers/page_scroll_controller.js

```diff
@@ -2,14 +2,8 @@ import { Controller } from "@hotwired/stimulus"
 
 export default class extends Controller {
   connect() {
-    document.addEventListener("turbo:load", this.scrollIntoView)
-  }
-
-  disconnect() {
-    document.removeEventListener("turbo:load", this.scrollIntoView)
-  }
-
-  scrollIntoView() {
-    this.element.scrollIntoView(false)
+    setTimeout(() => {
+      this.element.scrollIntoView(false)
+    }, 0);
```

Comment:
Rather than using a setTimeout I would suggest using the next available animation frame

```suggestion
    requestAnimationFrame(() => {
      this.element.scrollIntoView(false)
    })
```

From my testing it does now scroll into view but we are loosing focus on the textarea after submiting a message. For chat message I would expect focus to be retained.

here is what I ended with to maitain focus

```js
import { Controller } from "@hotwired/stimulus"

export default class extends Controller {
  static values = { 
    onConnect: {type: Boolean, default: true},
    focus: {type: Boolean, default: true},
  }                                

  connect() {
    requestAnimationFrame(() => {
      if (this.onConnectValue) this.scrollIntoView()
      if (this.focusValue) this.focus()
    })
  }

  // actions

  scrollIntoView() {
    this.element.scrollIntoView(false)
  }

  focus() {
    // focus on the first input , select or textarea in the element
    this.element.querySelector("input, select, textarea").focus()
  }
}
```

Reply:
This works well! Great suggestion, didn't think through to test focus after a submission. I will add the rest of the code on the bottom